### PR TITLE
Added possibility to compile jit code with llvm's optimization levels O1-O3

### PR DESCRIPTION
**Pull request by Garra1980**
This is an attempt to allow choosing llvm's optimization level for compiling JIT code. In some case this could give good performance boost (20-30% on some tpc-h queries)

Targeting #392

### CODE REVIEW

File: QueryEngine/CompilationOptions.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 enum class ExecutorDeviceType { CPU, GPU };
 
-enum class ExecutorOptLevel { Default, LoopStrengthReduction, ReductionJIT };
+enum class ExecutorOptLevel { Default, O1, O2, O3, LoopStrengthReduction, ReductionJIT };
```
Comment (shtilman) [inline QueryEngine/CompilationOptions.h]: Need a comment saying that O1, O2, O3 shouldn't move.
Or should assign values explicitly O1=1, O2=2, O3=3

File: QueryEngine/NativeCodegen.cpp [hunk 2]
```diff
@@ -100,23 +101,34 @@ void optimize_ir(llvm::Function* query_func,
                  llvm::Module* module,
                  const std::unordered_set<llvm::Function*>& live_funcs,
                  const CompilationOptions& co) {
-  llvm::legacy::PassManager pass_manager;
+  if (co.opt_level_ >= ExecutorOptLevel::O1 && co.opt_level_ <= ExecutorOptLevel::O3) {
```
Comment (shtilman) [inline QueryEngine/NativeCodegen.cpp]: Could we pick a consistent naming theme in this function? mpm, pass_manager..

File: QueryEngine/CompilationOptions.h (hunk 1 above)
Comment (Garra1980) [reply QueryEngine/CompilationOptions.h]: Done

File: QueryEngine/NativeCodegen.cpp (hunk 2 above)
Comment (Garra1980) [reply QueryEngine/NativeCodegen.cpp]: Done

**Pull request closed without merging**